### PR TITLE
[CGPROD-2767] "Manage" menu

### PR DESCRIPTION
**Pull request by BeeGodwin**
- Added a `context` argument to scrollable list constructor, which points it at a collection to use
- Added `collections` config to provide contexts for shop and manage lists
- Home screen now loads the inventory collection (i.e. we expect it to be globally available) 
- Shop screens specify the collection they will sell from
- Title changes when moving between menus (work still to be done on title layout rules, so it looks 💩 for now)
- Scrollable list buttons now acquire the appropriate overlays for shop and manage menus
- Scrollable list buttons toggle overlays appropriately (from showing the price to 'owned', and from 'equip' to 'equipped')
- Some deduplication of config keys

### CODE REVIEW

File: src/components/shop/shop.js [hunk 1]
```diff
@@ -69,6 +69,7 @@ export class Shop extends Screen {
         Object.keys(this.panes).forEach(key =>
             pane === key ? this.panes[key].setVisible(true) : this.panes[key].setVisible(false),
         );
+        this.title.setTitleText(pane === "top" ? "Shop" : pane);
```
Comment (BeeGodwin) [inline src/components/shop/shop.js:72]: That hard-coded string probably needs to go, but have left this for future tix happening around the titles.

File: src/components/shop/balance-ui.js [hunk 2]
```diff
@@ -17,7 +17,7 @@ const styleDefaults = {
 const makeElement = makerFns => conf => makerFns[conf.type](conf).setOrigin(0.5);
 
 export const createBalance = (scene, metrics, safeArea) => {
-    const image = conf => scene.add.image(0, 0, `${scene.config.assetKeys.prefix}.${conf.key}`);
+    const image = conf => scene.add.image(0, 0, `${scene.config.assetPrefix}.${conf.key}`);
```
Comment (AdamXA) [inline src/components/shop/balance-ui.js:20]: can just do `scene.assetPrefix` I think

File: src/components/shop/menu-buttons.js [hunk 3]
```diff
@@ -59,12 +59,12 @@ const getButtonPosition = (containerBounds, idx, yOffset) => {
...
-const assetKey = (key, config) => `${config.assetKeys.prefix}.${config.assetKeys[key]}`;
+const assetKey = (sceneConfig, key, config) => `${sceneConfig.assetPrefix}.${config.assetKeys[key]}`;
 
 const getScale = (containerBounds, button) => containerBounds.width / button.width;
 
 const setButtonOverlays = (scene, button, title, config) => {
     const offset = button.width / 4;
     button.overlays.set("caption", scene.add.text(-offset / 2, 0, title, { ...styleDefaults }).setOrigin(0, 0.5));
-    button.overlays.set("icon", scene.add.image(-offset, 0, assetKey("buttonIcon", config)));
+    button.overlays.set("icon", scene.add.image(-offset, 0, assetKey(scene.config, "buttonIcon", config)));
```
Comment (AdamXA) [inline src/components/shop/menu-buttons.js:69]: can just pass scene and do `scene.assetPrefix`

File: src/core/layout/scrollable-list/scrollable-list-buttons.js [hunk 4]
```diff
@@ -22,13 +24,26 @@ const createGelButton = (scene, item) => {
         group: scene.scene.key,
         id,
         key: config.assetKeys.itemBackground,
-        scene: config.assetKeys.prefix,
+        scene: config.assetPrefix,
```
Comment (AdamXA) [inline src/core/layout/scrollable-list/scrollable-list-buttons.js:27]: guess scene isnt the actual scene :D

also bit confused as why theres scene.key and scene.assetPrefix being used here

File: src/core/layout/scrollable-list/scrollable-list.js [hunk 5]
```diff
@@ -101,9 +103,9 @@ const setupEvents = (scene, panel) => {
 const getPanelItems = panel => panel.getByName("grid", true).getElement("items");
 
 export class ScrollableList extends Phaser.GameObjects.Container {
-    constructor(scene) {
+    constructor(scene, context) {
```
Comment (AdamXA) [inline src/core/layout/scrollable-list/scrollable-list.js:106]: guessing `context` here is a string like "shop" or "manage"

the word `context` might get confused with something like `screen.context`

File: src/core/layout/scrollable-list/scrollable-list.js [hunk 6]
```diff
@@ -57,15 +57,17 @@ const createTable = scene => {
         name: "grid",
     });
 
-    collection.forEach((item, idx) => table.add(createItem(scene, item), 0, idx, "top", 0, true));
+    const initState = "cta"; // PH
```
Comment (AdamXA) [inline src/core/layout/scrollable-list/scrollable-list.js:60]: 🤔

File: src/core/layout/scrollable-list/scrollable-list-buttons.js [hunk 7]
```diff
@@ -22,13 +24,26 @@ const createGelButton = (scene, item) => {
...
+            options: config.overlay.options[context],
+        },
+        setAll: setOverlays(gelButton, item),
+        unsetAll: unsetOverlays(gelButton),
+        state: STATES.find(st => st === state),
+        toggleState: toggleState(gelButton),
+        toggle: toggle(gelButton),
+    };
+
+    const callback = () => gelButton.overlays.toggle();
```
Comment (FostUK) [inline src/core/layout/scrollable-list/scrollable-list-buttons.js:46]: const callback = gelButton.overlays.toggle;

should work?

**Review by FostUK — APPROVED** [review]
Looks great - did some super minor changes but might be worth checking the diff between yours just so you know what's changed.

Gonna merge now!

image

**Pull request merged**